### ISSUE TITLE
bug: crash when editing a messages

### ISSUE THREAD

**Issue by thiswillbeyourgithub** [AndraxDev/speak-gpt#127]
Hi, just after editing a message the assistant crashes. Here's the error message:

```

App has been crashed and needs to be restarted.

===== BEGIN SYSTEM INFO =====
App version: 4.10 (410)
===== END SYSTEM INFO =====

===== BEGIN OF CRASH =====
java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
	at java.util.Objects.checkIndex(Objects.java:359)
	at java.util.ArrayList.get(ArrayList.java:434)
	at db.e0.onClick(Unknown Source:157)
	at c.i.handleMessage(Unknown Source:37)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8532)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:359)

===== END OF CRASH =====
```

**Comment by thiswillbeyourgithub** [comment]
Sure!

1. Open assistant
2. Manually enter "hi" and press send (note: also confirmed if using the microphone)
3. Wait for LLM to respond fully (note: actually the crash can be reproduced even if I cancel the llm response before it finished)
4. Open the edit window of the first message (so user message that said "hi") (note: also confirmed with assistant message, as well as 3rd or 4rth message in a longer chat)
5. Click "save" (you don't even have to change the content of the message, you can press save directly)
6. Notice the crash


It does not happen outside the assistant.

**Comment by AndraxDev** [comment]
Tried it now and I had no crash. Do you have this error every time you edit the message?

**Comment by thiswillbeyourgithub** [comment]
Yes.

Additional informations:
* When I edit "Hi" -> "whatever", it crashes but when I open the saved chats I see that 1 the chat was saved and 2 the message is still "Hi"
* I have auto send to off
* I use the "classic chat layout", the "new UI", "auto save chats", have a system message and no activation prompt
* I also get the crash on my tablet : Android 12 with version 4.10

**Comment by thiswillbeyourgithub** [comment]
Also, i get the crash when deleting a message, heres the log:

```
===== BEGIN OF CRASH =====
java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 0
	at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
	at java.util.Objects.checkIndex(Objects.java:359)
	at java.util.ArrayList.remove(ArrayList.java:511)
	at bb.k.c(Unknown Source:57)
	at db.e0.onClick(Unknown Source:19)
	at c.i.handleMessage(Unknown Source:37)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8532)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:359)

===== END OF CRASH =====

```

**Comment by AndraxDev** [comment]
App has been tested on the following devices: Real devices: Pixel 8, Pixel 8 Pro, Pixel Fold, Samsung Galaxy S22, Samsung Galaxy S23, Poco X3 Poco X6, Lenovo F10 tablet, emulators: Pixel 7, Pixel 6, Pixel 5, Window Subsystem for Android, android versions: 10, 11, 12, 13, 14, 15, heavily modified roms and pure android.

Result: None of these devices have this bug (nor in chat nor in assistant).

I will wait for other user reports with the same problem. As for now I can not fix this crash because I can not confirm that it really exists.

**Comment by AndraxDev** [comment]
If no other users or me detect this bug I will have to close this issue as not planned.

**Comment by AndraxDev** [comment]
Proofs:

https://github.com/AndraxDev/speak-gpt/assets/43646136/7a64aab0-9fa4-4588-a824-368d07eaa7d0

![Screenshot_2024-05-29-20-06-13-953_org.teslasoft.assistant.jpg](https://github.com/AndraxDev/speak-gpt/assets/43646136/18d6031b-6559-4278-a107-ff36834d984c)

**Comment by AndraxDev** [comment]
Fixed in SpeakGPT 4.11
